api/src/main/java/com/graphhopper/util/CustomModel.java

```diff
@@ -126,43 +136,38 @@ public String toString() {
 
     private String createContentString() {
         // used to check against stored custom models, see #2026
-        return "distanceInfluence=" + distanceInfluence + "|speedFactor=" + speedFactorMap + "|maxSpeed=" + maxSpeedMap +
-                "|maxSpeedFallback=" + maxSpeedFallback + "|priorityMap=" + priorityMap + "|areas=" + areas;
+        return "distanceInfluence=" + distanceInfluence + "|headingPenalty=" + headingPenalty
+                + "|speedStatements=" + speedStatements + "|priorityStatements=" + priorityStatements + "|areas=" + areas;
     }
 
     /**
      * A new CustomModel is created from the baseModel merged with the specified queryModel.
      */
     public static CustomModel merge(CustomModel baseModel, CustomModel queryModel) {
-        // avoid changing the specified CustomModel via deep copy otherwise the server-side CustomModel would be modified (same problem if queryModel would be used as target)
+        if (queryModel.isInternal())
+            throw new IllegalArgumentException("CustomModel in query cannot be internal");
+
+        // avoid changing the specified CustomModel via deep copy otherwise the server-side CustomModel would be
+        // modified (same problem if queryModel would be used as target)
         CustomModel mergedCM = new CustomModel(baseModel);
-        if (queryModel.maxSpeedFallback != null) {
-            if (mergedCM.maxSpeedFallback != null && mergedCM.maxSpeedFallback > queryModel.maxSpeedFallback)
-                throw new IllegalArgumentException("CustomModel in query can only use max_speed_fallback bigger or equal to " + mergedCM.maxSpeedFallback);
-            mergedCM.maxSpeedFallback = queryModel.maxSpeedFallback;
-        }
+        // we only overwrite the distance influence if a non-default value was used
         if (Math.abs(queryModel.distanceInfluence - CustomModel.DEFAULT_D_I) > 0.01) {
```

Comment:
But what if we set `distance_influence` to some value on the server side (say 50) and we want to set it to DEFAULT_D_I in a query? And can we rename `DEFAULT_D_I` to `DEFAULT_DISTANCE_INFLUENCE`? `_D_I` looks so cryptic.

Reply:
Renamed it.

> But what if we set distance_influence to some value on the server side (say 50) and we want to set it to DEFAULT_D_I in a query? 

Ah, we probably need a `Double` instead of `double` that is `null` if not set in the query?

Reply:
Yes with a Double we can better recognize the value was 'not given'.